### PR TITLE
Fix docker actions

### PR DESCRIPTION
**Pull request by mikiquantum**
Ensure that image is built and pushed only on commits to master.

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      - uses: cachix/cachix-action@v8
+        with:
+          name: centrifuge-chain
+          authToken: "${{ secrets.CACHIX_AUTH_TOKEN }}"
+      - run: nix build -L .#dockerContainer
+      - run: docker load -i result
+      - name: Extract branch name
+        shell: bash
+        run: echo "##[set-output name=branch;]$(echo ${GITHUB_REF##*/})"
+        id: extract_branch
```
Comment (asymmetric) [inline .github/workflows/docker.yml:28]: Are we using this for something?

Comment (mikiquantum) [reply .github/workflows/docker.yml:28]: Not anymore, we can get rid of it.

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+name: docker
+on:
+  push:
+    branches: [master]
```
Comment (asymmetric) [inline .github/workflows/docker.yml:4]: @mikiquantum should we instead do this only when tags are pushed? We don't necessarily want to push a Docker tag on every commit that is merged into master (think of a PR only changing CI-related files for examples).

Comment (asymmetric) [reply .github/workflows/docker.yml:4]: We've decided not to do this.

**Pull request merged**